src/get-imports.ts

```diff
@@ -0,0 +1,34 @@
+import { jsxToAst } from './modules/jsx';
+import { allText } from './editor';
+import traverse from '@babel/traverse';
+import { File } from '@babel/types';
+import { transformFromAstSync } from '@babel/core';
+
+export default function getImports(code: string): string {
+  let imports: Array<string> = [];
+  let sourceAst: File = jsxToAst(allText());
+  sourceAst.program.body = sourceAst.program.body.filter(bodyItem => bodyItem.type === 'ImportDeclaration');
+
+  const visitor = {
+    JSXIdentifier(path) {
+      if (path.container.type === 'JSXOpeningElement') {
+        imports = [...imports, path.node.name];
+      }
+    },
+  };
+
+  // include React from Source
+  imports = [...imports, 'React'];
```

Comment:
@borislit React is added to the imports on the exported components imports here as it (react) wouldn't be picked up in the selected codes JSXOpeningElements

Reply:
Just for clarification, this function search the selected code for JSXOpeningElements. Then searches that files' imports for the found JSXElements